### PR TITLE
[Bugfix] Ensure notification payload is passed to completion handler

### LINKED ISSUE #10863: remote push notification broken in 0.38-rc (with fix)
Looks like a change concerning the completion handler has broken the body on the push notification that gets delivered to the js code.  I think this is the fix:

https://github.com/facebook/react-native/pull/10860

### PR DESCRIPTION
**Pull request by npomfret**
Fixes #10863

### CODE REVIEW

File: Libraries/PushNotificationIOS/RCTPushNotificationManager.m [hunk 1]
```diff
@@ -219,7 +219,7 @@ - (void)handleRemoteNotificationReceived:(NSNotification *)notification
     self.remoteNotificationCallbacks[notificationId] = completionHandler;
   }
 
-  [self sendEventWithName:@"remoteNotificationReceived" body:notification.userInfo];
+  [self sendEventWithName:@"remoteNotificationReceived" body:remoteNotification];
```
Comment (JAStanton) [inline Libraries/PushNotificationIOS/RCTPushNotificationManager.m]: 👍  this is a result of a 9 month long PR with subtle refactors over long periods of time and not tested after each refactor. Thank you for catching this, I believe what you have here is correct

Comment (npomfret) [inline Libraries/PushNotificationIOS/RCTPushNotificationManager.m]: Cool.  will it go into 38?

Comment (JAStanton) [inline Libraries/PushNotificationIOS/RCTPushNotificationManager.m]: ¯\\_(ツ)_/¯ I'm a contributor not an owner, I don't have merge control :)

**Comment by hramos** [comment]
Can you describe what problem this PR is addressing?

**Comment by npomfret** [comment]
Sure,

RN0.38 introduces a new feature that allows the user to call an iOS completion handler after processing a remote push notification.  It's a great feature, without it there's no way to tell the OS that you've done the work to process a push notification. The OS will penalise your app and will eventually stop delivering push notifications.

I think some refactor wasn't quite tested and the push notification object that gets delivered to the RN  app is of the wrong type.  In dev mode you get a red error screen and the notification object doesn't have the expected data in it.  Basically, all iOS remote push notifications are broken in 0.38-rc. 

This fix results in the correct push notification object getting delivered to the app.  Its just a one-liner - should be easy for someone to test.

**Comment by JAStanton** [comment]
`- should be easy for someone to test.`
@npomfret have you tested this change to make sure it works after your change?

**Comment by npomfret** [comment]
Yes i tested (of course!).  Before the change, when an app receives a remote push notification you get a red error screen and the notification object delivered to the JS code doesn't contain the notification data.  After the change it works as expected.

I just meant it would be easy someone else to check...  my first rn PR you see.

**Comment by hramos** [comment]
No worries @npomfret, thanks for submitting the fix! We're just getting some more additional information to help us understand the scope and impact of the PR.

**Comment by hramos** [comment]
@facebook-github-bot shipit

**Comment by hramos** [comment]
Ah - @npomfret - just one more thing to do before we can import this.

**Comment by JAStanton** [comment]
@npomfret Thanks again, I feel foolish for letting this slip by, thanks for testing it and fixing it :)

**Comment by hramos** [comment]
Waiting on CLA to be signed before we can import this.

**Comment by npomfret** [comment]
What's a CLA?

**Comment by ide** [comment]
See the second reply: https://code.facebook.com/cla

**Comment by npomfret** [comment]
Hi, I just checked the [38 branch](https://github.com/facebook/react-native/blob/0.38-stable/Libraries/PushNotificationIOS/RCTPushNotificationManager.m#L222) and this PR still isn't present.  Was wondering if there's a hold-up. Is there anything else I need to do to get it in there?

(Sorry, not familiar with the workflow here yet)

**Comment by grabbou** [comment]
Nope @npomfret, cherry-picking it now and releasing stable :)

**Pull request closed without merging**